base/logging.jl

```diff
@@ -369,9 +375,10 @@ function current_logstate()
 end
 
 # helper function to get the current logger, if enabled for the specified message type
-@noinline function current_logger_for_env(std_level::LogLevel, group, _module)
+function current_logger_for_env(std_level::LogLevel, group, _module)
```

Comment:
We don't want this to inline. It'll bloat code everywhere, but won't improve performance

Reply:
Tested with BenchmarkTools, removing `@noinline` is faster, 75 vs 80 ns. Isn't the main bloat in `env_override_minlevel` anyway?

Reply:
Of course it'll look faster on a synthetic benchmark – it's doing strictly less work, assuming all of your code fits in the processor cache. I'm surprised the `hasenv` check here makes a difference though. That's essentially all that env_override_minlevel is doing, maybe it should have an explicit `isempty(debug) && return`, in case codegen isn't able to handle our `x == ""` call?

Reply:
I tested with a slightly less synthetic benchmark, but OK I get your point.
I had assumed a difference between haskey and getindex; I'll have a closer look.